mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/chess/BoardWithHistory.kt

```diff
@@ -0,0 +1,35 @@
+package ch.epfl.sdp.mobile.test.application.chess
+
+import ch.epfl.sdp.mobile.application.chess.Board
+import ch.epfl.sdp.mobile.application.chess.BoardWithHistory
+
+/**
+ * An implementation of [BoardWithHistory] which can be used to build some fake histories of games.
+ *
+ * @param Piece the type of the pieces.
+ */
+private data class FakeBoardWithHistory<Piece : Any>(
+    val board: Board<Piece>,
+    private val previous: FakeBoardWithHistory<Piece>?
+) : BoardWithHistory<Piece>, Board<Piece> by board {
```

Comment:
What does the `by` here, isn't this a class ?

Reply:
https://kotlinlang.org/docs/delegation.html

Reply:
💋